commit cb3fcb59c81d9fb050df3b8d552c27b6e5651648
Author: Francis Whittle <francis@agileware.com.au>
Date:   Mon Feb 25 11:13:18 2013 +1100

    uc_discounts_alt: check existence of order variables when testing them

diff --git a/uc_discounts/uc_discounts.module b/uc_discounts/uc_discounts.module
index 98c0cac..2bbf569 100644
--- a/uc_discounts/uc_discounts.module
+++ b/uc_discounts/uc_discounts.module
@@ -252,7 +252,8 @@ function uc_discounts_uc_order($op, &$arg1, $arg2) {
 
     case 'save':
       //If discount line items need updating
-      if ($arg1->uc_discounts_line_items_need_updating) {
+      //Use empty() as not always set
+      if (!empty($arg1->uc_discounts_line_items_need_updating)) {
         //Delete existing order codes
         uc_discounts_order_codes_delete($arg1->order_id);
 
@@ -289,10 +290,12 @@ function uc_discounts_uc_order($op, &$arg1, $arg2) {
         }
 
         //Add discount line items
-        foreach ($arg1->uc_discounts_line_items as $line_item) {
-          if ($line_item['amount'] != 0) {
-            uc_order_line_item_add($arg1->order_id, $line_item['type'], $line_item['title'], $line_item['amount'], $line_item['weight'], $line_item['data']);
-            $new_order_line_items[] = $line_item;
+        if(!empty($arg1->uc_discounts_line_items)) {
+          foreach ($arg1->uc_discounts_line_items as $line_item) {
+            if ($line_item['amount'] != 0) {
+              uc_order_line_item_add($arg1->order_id, $line_item['type'], $line_item['title'], $line_item['amount'], $line_item['weight'], $line_item['data']);
+              $new_order_line_items[] = $line_item;
+            }
           }
         }
 
